protocol/app/app.go

```diff
@@ -1460,18 +1429,170 @@ func (app *App) initSlinkySidecarClient(appOpts servertypes.AppOptions) oraclecl
+	if app.oracleMetrics == nil {
+		app.oracleMetrics = app.initOracleMetrics(appOpts)
+	}
+	cfg, err := oracleconfig.ReadConfigFromAppOpts(appOpts)
+	if err != nil {
+		panic(err)
+	}
 	slinkyClient, err := oracleclient.NewClientFromConfig(
 		cfg,
 		app.Logger().With("client", "oracle"),
-		oracleMetrics,
+		app.oracleMetrics,
 	)
 	if err != nil {
 		panic(err)
 	}
 	return slinkyClient
 }
 
+func (app *App) createProposalHandlers(
+	appFlags flags.Flags,
+	txConfig client.TxConfig,
+	appOpts servertypes.AppOptions,
+) (sdk.PrepareProposalHandler, sdk.ProcessProposalHandler) {
+	strategy := currencypair.NewDefaultCurrencyPairStrategy(app.PricesKeeper)
+	veCodec := compression.NewCompressionVoteExtensionCodec(
+		compression.NewDefaultVoteExtensionCodec(),
+		compression.NewZLibCompressor(),
+	)
+	extCommitCodec := compression.NewCompressionExtendedCommitCodec(
+		compression.NewDefaultExtendedCommitCodec(),
+		compression.NewZLibCompressor(),
+	)
+
+	var priceUpdateGenerator prices.PriceUpdateGenerator = prices.NewDefaultPriceUpdateGenerator(app.PricesKeeper)
+	var priceUpdateDecoder process.UpdateMarketPriceTxDecoder = process.NewDefaultUpdateMarketPriceTxDecoder(
+		app.PricesKeeper, app.txConfig.TxDecoder())
+
+	if appFlags.VEOracleEnabled {
+		priceUpdateGenerator = prices.NewSlinkyPriceUpdateGenerator(
+			aggregator.NewDefaultVoteAggregator(
+				app.Logger(),
+				voteweighted.MedianFromContext(
+					app.Logger(),
+					app.StakingKeeper,
+					voteweighted.DefaultPowerThreshold,
+				),
+				strategy,
+			),
+			extCommitCodec,
+			veCodec,
+			strategy,
+		)
+		priceUpdateDecoder = process.NewSlinkyMarketPriceDecoder(
+			priceUpdateDecoder,
+			priceUpdateGenerator,
+		)
+	}
+	dydxPrepareProposalHandler := prepare.PrepareProposalHandler(
+		txConfig,
+		app.BridgeKeeper,
+		app.ClobKeeper,
+		app.PerpetualsKeeper,
+		priceUpdateGenerator,
+	)
+	// if the node is a NonValidatingFullNode, we don't need to run any of the oracle code
+	if appFlags.NonValidatingFullNode {
+		dydxPrepareProposalHandler = prepare.FullNodePrepareProposalHandler()
+	} else {
+		if app.oracleMetrics == nil {
+			app.oracleMetrics = app.initOracleMetrics(appOpts)
+		}
+		app.initOracle(priceUpdateDecoder)
+	}
+	if app.oracleMetrics == nil {
+		app.oracleMetrics = servicemetrics.NewNopMetrics()
+	}
+
+	// ProcessProposal setup.
+	var dydxProcessProposalHandler sdk.ProcessProposalHandler
+	if appFlags.NonValidatingFullNode {
+		// Note: If the command-line flag `--non-validating-full-node` is enabled, this node will use
+		// an implementation of `ProcessProposal` which always returns `abci.ResponseProcessProposal_ACCEPT`.
+		// Full-nodes do not participate in consensus, and therefore should not participate in voting / `ProcessProposal`.
+		dydxProcessProposalHandler = process.FullNodeProcessProposalHandler(
+			txConfig,
+			app.BridgeKeeper,
+			app.ClobKeeper,
+			app.StakingKeeper,
+			app.PerpetualsKeeper,
+			priceUpdateDecoder,
+		)
```

Comment:
if NonValidatingFullNode and VEOracleEnabled, right now you are returning the VE process proposal handler instead of the full node process proposal handler. Shouldn't we `return dydxPrepareProposalHandler, dydxProcessProposalHandler` here?